freud/box.pyx

```diff
@@ -921,6 +940,75 @@ cdef class Box:
                             "positional argument: L")
         return cls(Lx=L, Ly=L, Lz=0, xy=0, xz=0, yz=0, is2D=True)
 
+    @classmethod
+    def from_lattice_vectors(cls, lattice_vectors, dimensions: int = None):
+        """Create a unit cell from lattice vectors.
+
+        Args:
+            lattice_vectors (array-like):
+                The lattice vectors. The dimensions of the object should be 3x3. Lattice
+                vector a1 is lattice_vectors[0], etc.
+            dimensions (int): The number of dimensions (Default value = :code:`None`)
+
+        Returns:
+            :class:`~.UnitCell`: A unit cell with the given lattice vectors.
+        """
+        lattice_matrix = np.asarray(lattice_vectors, dtype=np.float32)
+        v0 = lattice_matrix[0]
+        v1 = lattice_matrix[1]
+        v2 = lattice_matrix[2]
+        Lx = np.sqrt(np.dot(v0, v0))
+        a2x = np.dot(v0, v1) / Lx
+        Ly = np.sqrt(np.dot(v1, v1) - a2x * a2x)
```

Comment:
```suggestion
        Ly = np.sqrt(np.dot(v1, v1) - a2x**2)
```

I think a square here better conveys the intent, as below we have a very similar function with two different variables `(a2x * a3x)`